dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentReportHelper.java

```diff
@@ -0,0 +1,203 @@
+package com.dotcms.rest.api.v1.content;
+
+import com.dotcms.contenttype.business.ContentTypeAPI;
+import com.dotcms.contenttype.model.type.BaseContentType;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.exception.ExceptionUtil;
+import com.dotcms.util.pagination.ContentReportPaginator;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.common.util.SQLUtil;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.HostAPI;
+import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UUIDUtil;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import io.vavr.Lazy;
+import org.apache.commons.lang3.math.NumberUtils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * This helper class provides utility methods to generate Content Reports for different dotCMS
+ * objects.
+ * <p> Each dotCMS object may have its own way to generate the Content Report. For instance, the
+ * report for a Site uses the Content Type API, whereas the report for a Folder uses the Folder API
+ * to retrieve the required data. You just need to make sure that the implementation you add for a
+ * specific dotCMS object transforms its data into the expected {@link ContentReportView}
+ * object.</p>
+ *
+ * @author Jose Castro
+ * @since Mar 7th, 2024
+ */
+public class ContentReportHelper {
```

Comment:
I would make this an abstract class with a generic `getContentReport` method that's overridden by the two extenders SiteContentReportHelper and FolderContentReportHelper and move the private methods to these extender classes accordingly.

Reply:
Done!

Reply:
I think interfaces play better pattern here than abstract classes

Reply:
Done!